lib/src/syn/v1/operator.rs

```diff
@@ -132,11 +135,28 @@ pub fn matches(i: &str) -> IResult<&str, Operator> {
 	})(i)
 }
 
+pub fn knn_distance(i: &str) -> IResult<&str, Distance> {
+	let (i, _) = char(',')(i)?;
+	alt((
+		map(tag_no_case("CHEBYSHEV"), |_| Distance::Chebyshev),
+		map(tag_no_case("COSINE"), |_| Distance::Cosine),
+		map(tag_no_case("EUCLIDEAN"), |_| Distance::Euclidean),
+		map(tag_no_case("HAMMING"), |_| Distance::Hamming),
+		map(tag_no_case("JACCARD"), |_| Distance::Jaccard),
+		map(tag_no_case("MANHATTAN"), |_| Distance::Manhattan),
+		minkowski,
+		map(tag_no_case("PEARSON"), |_| Distance::Pearson),
+	))(i)
+}
+
 pub fn knn(i: &str) -> IResult<&str, Operator> {
 	let (i, _) = char('<')(i)?;
-	let (i, k) = u32(i)?;
-	let (i, _) = char('>')(i)?;
-	Ok((i, Operator::Knn(k)))
+	cut(|i| {
+		let (i, k) = u16(i)?;
+		let (i, dist) = opt(knn_distance)(i)?;
+		let (i, _) = char('>')(i)?;
+		Ok((i, Operator::Knn(k, dist)))
+	})(i)
```

Comment:
Can you cut here? It seems to me that when the parser encounters something like `1 < 2` this would cause the parser to fail early because it can't find the second `>` for the knn operator it expects.

Reply:
This seems to also be cause for the test failing to complete.